traveler_stripe_payment_provider/src/main/java/com/guestlogix/traveler_stripe_payment_provider/StripePaymentManager.java

```diff
@@ -0,0 +1,68 @@
+package com.guestlogix.traveler_stripe_payment_provider;
+
+import android.content.Context;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.guestlogix.travelercorekit.callbacks.PaymentsFetchCallback;
+import com.guestlogix.travelercorekit.models.Payment;
+import com.guestlogix.travelercorekit.models.PaymentManager;
+import com.stripe.android.CustomerSession;
+import com.stripe.android.StripeError;
+import com.stripe.android.model.PaymentMethod;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class StripePaymentManager implements PaymentManager {
+    Context context;
+
+    public StripePaymentManager(Context context) {
+        this.context = context;
+    }
+
+    @Override
+    public void fetchPayments(PaymentsFetchCallback callback) {
+        CustomerSession.initCustomerSession(context, new TravelerEphemeralKeyProvider());
+        CustomerSession.getInstance().getPaymentMethods(PaymentMethod.Type.Card, new CustomerSession.PaymentMethodsRetrievalListener() {
+            @Override
+            public void onPaymentMethodsRetrieved(@NonNull List<PaymentMethod> paymentMethods) {
+                ArrayList<Payment> payments = new ArrayList<>();
+
+                for (PaymentMethod method: paymentMethods) {
+                    payments.add(new StripePayment(method));
+                }
+
+                callback.onPaymentsFetchSuccess(payments);
+            }
+
+            @Override
+            public void onError(int errorCode, @NonNull String errorMessage, @Nullable StripeError stripeError) {
+                callback.onPaymentsFetchError(new Error(errorMessage));
+            }
+        });
+    }
+
+    @Override
+    public void savePayment(Payment payment) {
```

Comment:
would it make sense for savePayment to take in a CustomerSession.PaymentMethodRetrievalListener()?